esx_service/utils/auth.py

```diff
@@ -91,34 +91,34 @@ def get_default_tenant():
         return None, None, None
 
 
-def get_default_privileges():
-    """ Get DEFAULT privileges by querying the auth DB.
-        DEFAULT privilege will match any (tenant_uuid, datastore) pair which
-        does not have an entry in privileges table explicitly
+def get_all_ds_privileges(tenant_uuid):
+    """ Get privileges on _ALL_DS by querying the auth DB.
+        For the tenant with given tenant_uuid, this privilege
+        will match any datastore which does not have an entry
+        in privileges table explicitly
```

Comment:
We do have this privilege to "__ALL_DS" in the privilege table.  For example,  for a given tenant "tenant1", user added a privilege to "datastore1" , then in the privilege table, a entry with <tenant1, datastore1> will be added. So any request from "tenant1" to ask checking authorization to "datastore1" will match this privilege. If we have the "__ALL_DS" privilege for "tenant1", in the privilege table, an entry with <tenant1, __ALL_DS> will be added. Any request from "tenant1" to ask checking authorization to any datastore other than "datastore1" will be checked against this entry.